drivers/soundwire/cadence_master.c

```diff
@@ -1178,6 +1186,133 @@ static const struct sdw_master_port_ops cdns_port_ops = {
 	.dpn_port_enable_ch = cdns_port_enable,
 };
 
+/**
+ * sdw_cdns_suspend: Cadence PM suspend routine
+ *
+ * @cdns: Cadence instance
+ */
+int sdw_cdns_suspend(struct sdw_cdns *cdns, bool block_wake)
+{
+	u32 status;
+	int ret;
+
+	/* Check suspend status */
+	status = cdns_readl(cdns, CDNS_MCP_STAT);
+	if (status & CDNS_MCP_STAT_CLK_STOP) {
+		dev_dbg(cdns->dev, "Clock is already stopped\n");
+		return 1;
+	}
+
+	/*
+	 * In some system it is required to be able to put master
+	 * into a state in which it ignores wake-up trials in clock
+	 * stop state
+	 */
+	if (block_wake)
+		cdns_updatel(cdns, CDNS_MCP_CONTROL,
+			     CDNS_MCP_CONTROL_BLOCK_WAKEUP,
+			     CDNS_MCP_CONTROL_BLOCK_WAKEUP);
+
+	/* Disable cmd accept mode */
+	cdns_updatel(cdns, CDNS_MCP_CONTROL, CDNS_MCP_CONTROL_CMD_ACCEPT, 0);
```

Comment:
sorry, the comments should be not exactly correct.  According to spec, we need to set CMD_ACCEPT to true to cause frame shape change or to enter clock stop if there is no slave attached.  And if there are some slaves attached, it should be set false.

BTW since default setting in cdns_init is true, we need to set it to false if there is a slave attached, and set it to true if there is no slave attached.  This is another issue.